src/libtrx/game/music/backend_cdaudio.c

```diff
@@ -40,7 +41,8 @@ static bool M_Parse(M_BACKEND_DATA *const data)
         return false;
     }
 
-    data->tracks = Memory_Alloc(sizeof(M_CDAUDIO_TRACK) * MAX_MUSIC_TRACKS);
+    data->num_tracks = track_content_size / sizeof(M_CDAUDIO_TRACK);
```

Comment:
I don't think this is right – `cdaudio.dat` is a text file, and while it may be a good enough heuristic I'd rather not combine these units. We'll likely need to iterate through the text once, extract max track ID, allocate and re-parse.

Reply:
Oops, so it is. I don't know why I thought it was flat binary 😄 Will adjust.

Reply:
Updated - we use a vector while parsing, then we allocate to the backend data before continuing as before with the logic to reindex as necessary.